packages/blazor-workspace/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js

```diff
@@ -162,12 +163,48 @@ export function afterStarted(Blazor) {
     });
 }
 
+function handleRuntimeStarted() {
+    window.NimbleBlazor.hasRuntimeStarted = true;
+}
+
+// Blazor Web Apps
+export function afterWebStarted(Blazor) {
+    registerNimbleEvents(Blazor);
+    // Note: For static SSR, this is the last event called, and hasRuntimeStarted
+    // will remain false.
```

Comment:
Why should hasRuntimeStarted remain false for static SSR?

Reply:
Revamped the startup script more, so I think this thread is resolved.

- A single `isInitialized` flag is exposed for C# to query for
- For static SSR, `isInitialized` will remain false. I didn't see a reason to add a method to toggle it from C# static SSR tests, since those tests won't be using the events anyway. `AcceptanceTestsBase` now defaults to skipping the initialization check entirely if the JS to evaluate isn't specified. So static SSR tests use the default behavior. Then `Spright/NimbleInteractiveAcceptanceTestsBase` add the JS (checking `isInitialized`).

Note: I did keep the event registration in `afterWebStarted`, as that's Microsoft's guidance for where to do it for Blazor Web Apps ([1](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling?view=aspnetcore-8.0#general-configuration), [2](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/startup?view=aspnetcore-8.0#javascript-initializers)).